server/src/main/scala/za/co/absa/atum/server/api/controller/PartitioningController.scala

```diff
@@ -32,6 +32,12 @@ trait PartitioningController {
     partitioningSubmitDTO: PartitioningSubmitV2DTO
   ): IO[ErrorResponse, (SingleSuccessResponse[PartitioningWithIdDTO], String)]
 
+  def patchPartitioningParentV2(
```

Comment:
My suggestion: try to create smaller PRs. In this case, all this could be split into 2 PRs:
* SQL stuff only, with Balta tests that would test proposed DB functions 
* Scala part related to the backend - controller, service, repository, scala classes related to DB function wrapping/calls, etc